mp4parse/src/lib.rs

```diff
@@ -1128,24 +1137,30 @@ fn read_esds<T: Read>(src: &mut BMFFBox<T>) -> Result<ES_Descriptor> {
     let esds_array = try!(read_buf(src, esds_size as usize));
 
     // Parsing DecoderConfig descriptor to get the object_profile_indicator
-    // for correct codec type.
-    let object_profile_indicator = {
+    // for correct codec type and audio sample rate.
+    let (object_profile_indicator, sample_frequency) = {
+        let mut object_profile: u8 = 0;
+        let mut sample_frequency = None;
+
         // clone a esds cursor for parsing.
         let esds = &mut Cursor::new(&esds_array);
-        let esds_tag = try!(esds.read_u8());
+        let mut next_tag = try!(esds.read_u8());
 
-        if esds_tag != ESDESCR_TAG {
+        if next_tag != ESDESCR_TAG {
             return Err(Error::Unsupported("fail to parse ES descriptor"));
         }
 
         let esds_extend = try!(esds.read_u8());
+        let mut esds_end: u64 = 0;
```

Comment:
Rather than turning off the warning with `#[allow(unused_assignments)]`, you can avoid it with an `if` expression like a ternary in C. I'm not sure this is actually easier to read, but avoiding the `mut` is more natural in rust, I think.

```rust
let esds_extend = try!(esds.read_u8());
// extension tag starts from 0x80.
let esds_end = if esds_extend >= 0x80 {
    // skip remaining extension.
    try!(skip(esds, 2));
    esds.position() + try!(esds.read_u8()) as u64
} else {
    esds.position() + esds_extend as u64
};
try!(skip(esds, 2));
```